client/src/pages/Profile.tsx

```diff
@@ -27,14 +32,58 @@ const fakeStudent: StudentProfileProps = {
   tutors: ['Steven', 'Joe'],
 };
 
+const initProfile: IProfile = {
+  id: -1,
+  firstName: 'Jane',
+  lastName: 'Doe',
+  age: 50,
+  bio: 'A great bio',
+  profileUrl: 'https://placeimg.com/192/192/people',
+  userId: -1,
+  email: 'janedoe@gmail.com',
+  role: Role.STUDENT,
+};
+
 function Profile() {
+  const [profile, setProfile] = useState<IProfile>(initProfile);
+  const [error, setError] = useState(false);
+  const params = useParams();
+
+  useEffect(() => {
+    const fetchProfile = async () => {
+      let profile;
+      try {
+        if (!params.id) {
+          profile = await getLoggedInUserProfile();
+        } else {
+          profile = await getProfile(params.id);
+        }
+
+        profile.students = fakeTutor.students;
+        profile.tutors = fakeStudent.tutors;
+
+        setProfile(profile);
+        setError(false);
+      } catch (err) {
+        console.log('Failed to load profile');
+        setError(true);
+      }
+    };
+
+    fetchProfile();
+  }, [params.id]);
+
   // Temporary filler until backend is integrated
   const role = 'TUTOR';
```

Comment:
We don't need this anymore, I just had that as filler while we were waiting for the data lol